### PR TITLE
Added data check action for setting first column as primary key if identified as ID

### PR DESCRIPTION
**Pull request by simha104**
### Pull Request Description
Added data check action for setting first column as primary key if identified as ID

### CODE REVIEW

File: evalml/data_checks/id_columns_data_check.py [hunk 1]
```diff
@@ -157,27 +186,50 @@ def validate(self, X, y=None):
...
+                warning_msg = (
+                    "The first column '{}' is {}% or more likely to be an ID column"
+                )
+                warning_msg = warning_msg.format(
+                    col_names[0],
+                    self.id_threshold * 100,
+                )
+                message_code = DataCheckMessageCode.HAS_ID_FIRST_COLUMN
+                action_code = DataCheckActionCode.SET_FIRST_COL_ID
+                details = {"column": col_names[0]}
```
Comment (ParthivNaresh) [inline evalml/data_checks/id_columns_data_check.py:209]: I think this should be changed to `columns`

File: evalml/data_checks/id_columns_data_check.py [hunk 2]
```diff
@@ -157,27 +186,50 @@ def validate(self, X, y=None):
...
+            else:
```
Comment (ParthivNaresh) [inline evalml/data_checks/id_columns_data_check.py:210]: Are we trying to make this an either/or scenario? I think we would still want to know if other columns could be ID columns right?

File: evalml/tests/data_checks_tests/test_id_columns_data_check.py [hunk 3]
```diff
@@ -69,15 +69,42 @@ def test_id_columns_warning():
...
+    X_dict = {
+        "col_1_id": [0, 1, 2, 3],
+        "col_2": [2, 3, 4, 5],
+        "col_3_id": [1, 1, 2, 3],
+        "Id": [3, 1, 2, 0],
+        "col_5": [0, 0, 1, 2],
+        "col_6": [0.1, 0.2, 0.3, 0.4],
+    }
+
+    X = pd.DataFrame.from_dict(X_dict)
```
Comment (ParthivNaresh) [inline evalml/tests/data_checks_tests/test_id_columns_data_check.py:95]: I think we would expect `col_3_id` to also show up under `DataCheckMessageCode.HAS_ID_COLUMN`.

File: evalml/tests/data_checks_tests/test_id_columns_data_check.py [hunk 4]
```diff
@@ -149,55 +216,55 @@ def test_id_cols_data_check_input_formats():
...
+                    metadata={"columns": [1]},
                 ),
             ],
         ).to_dict(),
     ]
 
     # test np.array
     assert id_cols_check.validate(
-        np.array([[0, 1], [1, 2], [2, 3], [3, 4], [4, 5]]),
+        np.array([[0, 1], [0, 2], [2, 3], [3, 4], [4, 5]]),
```
Comment (ParthivNaresh) [inline evalml/tests/data_checks_tests/test_id_columns_data_check.py:256]: Why were some of these other tests changed?

**Review by ParthivNaresh — COMMENTED** [review]
Thanks for the PR, I think we should expect other columns that could be IDs to also show up in the response alongside the potential primary key column

File: evalml/data_checks/data_check_action_code.py [hunk 5]
```diff
@@ -22,6 +22,9 @@ class DataCheckActionCode(Enum):
     REGULARIZE_AND_IMPUTE_DATASET = "regularize_and_impute_dataset"
     """Action code for regularizing and imputing all features and target time series data."""
 
+    SET_FIRST_COL_ID = "set_first_col_id"
+    """Action code for setting the first column as an id column"""
```
Comment (bchen1116) [inline evalml/data_checks/data_check_action_code.py:26]: nit: period

File: evalml/data_checks/data_check_message_code.py [hunk 6]
```diff
@@ -17,6 +17,9 @@ class DataCheckMessageCode(Enum):
     HAS_ID_COLUMN = "has_id_column"
     """Message code for data that has ID columns."""
 
+    HAS_ID_FIRST_COLUMN = "has_id_first_column"
+    """Message code for data that has an ID column as the first column"""
```
Comment (bchen1116) [inline evalml/data_checks/data_check_message_code.py:21]: nit: period

**Review by bchen1116 — APPROVED** [review]
Agreed with the comments Parthiv left. Left some nits, but otherwise looks good to me!

**Review by ParthivNaresh — APPROVED** [review]
Looks good!

**Pull request merged**